app/controllers/products_controller.rb

```diff
@@ -1,37 +1,34 @@
 class ProductsController < ApplicationController
-  before_action :require_login, except: [:index, :show, :merchant_products]
+  before_action :require_login, only: [:new, :create, :edit, :update, :destroy, :retire]
+  before_action :find_product,  only: [:show, :edit, :update, :destroy, :retire]
 
   def index
     @products = Product.active_product
   end
 
   def show
-    @product = Product.find(params[:id])
     @order_item = current_order.order_items.new
```

Comment:
This is an interesting use of the `new`. I'm curious why you used this method chain rather than directly on the `OrderItem.new`